tests/templates/test_state_preparations/test_state_prep_mps.py

```diff
@@ -21,232 +21,1581 @@
+
+        mps = [
+            np.array([[0.70710678, 0.0], [0.0, 0.70710678]]),
+            np.array(
+                [
+                    [[0.0, 1.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0]],
+                    [[0.0, 0.0, -0.0, 0.0], [-1.0, 0.0, 0.0, 0.0]],
+                ]
+            ),
+            np.array(
+                [
+                    [[0.00000000e00, 1.74315280e-32], [-7.07106781e-01, -7.07106781e-01]],
+                    [[7.07106781e-01, 7.07106781e-01], [0.00000000e00, 0.00000000e00]],
+                    [[0.00000000e00, 0.00000000e00], [-7.07106781e-01, 7.07106781e-01]],
+                    [[-7.07106781e-01, 7.07106781e-01], [0.00000000e00, 0.00000000e00]],
+                ]
+            ),
+            np.array([[1.0, 0.0], [0.0, 1.0]]),
+        ]
+
+        ops = qml.MPSPrep.compute_decomposition(mps, wires=range(2, 6), work_wires=[0, 1])
+
+        for ind, op in enumerate(ops):
+            assert op.wires == qml.wires.Wires([2 + ind] + [0, 1])
+            assert op.name == "QubitUnitary"
+
+    @pytest.mark.jax
+    def test_jax_jit_mps(self):
+        """Check the operation works with jax and jit."""
+
+        import jax
+        from jax import numpy as jnp
+
+        mps = [
+            jnp.array([[0.0, 0.107j], [0.994, 0.0]], dtype=complex),
+            jnp.array(
+                [
+                    [[0.0, 0.0], [1.0, 0.0]],
+                    [[0.0, 1.0], [0.0, 0.0]],
+                ],
+                dtype=complex,
+            ),
+            jnp.array([[-1.0, -0.0], [-0.0, -1.0]], dtype=complex),
+        ]
+
+        dev = qml.device("default.qubit")
+
+        @jax.jit
+        @qml.qnode(dev)
+        def circuit():
+            qml.MPSPrep(mps, wires=range(2, 5), work_wires=[0, 1])
+            return qml.state()
+
+        output = circuit()[:8]
+
+        state = [
+            0.0 + 0.0j,
+            -0.10705513j,
+            0.0 + 0.0j,
+            0.0 + 0.0j,
+            0.0 + 0.0j,
+            0.0 + 0.0j,
+            -0.99451217 + 0.0j,
+            0.0 + 0.0j,
+        ]
+
+        assert jax.numpy.allclose(output, jax.numpy.array(state), rtol=0.01)
+
+    def test_wires_decomposition(self):
+        """Checks that error is shown if no `work_wires` are given in decomposition"""
+
+        mps = [
+            np.array([[0.0, 0.107j], [0.994, 0.0]], dtype=complex),
+            np.array(
+                [
+                    [[0.0, 0.0], [1.0, 0.0]],
+                    [[0.0, 1.0], [0.0, 0.0]],
+                ],
+                dtype=complex,
+            ),
+            np.array([[-1.0, -0.0], [-0.0, -1.0]], dtype=complex),
+        ]
+
+        dev = qml.device("default.qubit")
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.MPSPrep(mps, wires=range(2, 5))
+            return qml.state()
+
+        with pytest.raises(AssertionError, match="The qml.MPSPREP decomposition requires"):
+            circuit()
```

Comment:
Why do we need a QNode/system test here? 

Following the [testing guidelines document](https://github.com/PennyLaneAI/guidance-docs/blob/master/development/testing-guidelines.md#writing-unit-tests) this could be replaced by a single unit test with a call to `decomposition()` , right
```suggestion
        op = qml.MPSPrep(mps, wires=range(2, 5))
        with pytest.raises(AssertionError, match="The qml.MPSPREP decomposition requires"):
            op.decomposition()
```